pkg/cache/loading.go

```diff
@@ -29,6 +29,8 @@ type LoadingLRUCacheWithTTL[K comparable, V any] struct {
 	lru    *LRUCacheWithTTL[K, V]
 	loader LoaderFunc[K, V]
 	closer func() error
+
+	sfg *singleflight.Group
```

Comment:
Why don't we create a new struct for `LoadingOnceCache` and put the singleflight there? It would be easier to reason about. We can embed the `LoadingLRUCacheWithTTL` into that struct and override the `Get` method. What do you think?

Reply:
Made the changes to do that, let me know what you think :)